components/AdminLessonsSideBar.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import { Lesson } from '../graphql/index'
+
+type SideBarLessonProps = {
+  lessons: Lesson[] | undefined
+  setLessons: React.Dispatch<React.SetStateAction<Lesson[] | null>>
+  setSelectedLesson: React.Dispatch<React.SetStateAction<number>>
+}
+
+type LessonProps = {
+  obj: { title: string }
+  i: number
+  setSelectedLesson: React.Dispatch<React.SetStateAction<number>>
+}
+
+const LessonTitle: React.FC<LessonProps> = ({ obj, setSelectedLesson, i }) => (
+  <div key={i} data-testid="challenge-title" className="card mb-2">
+    <div className="btn d-flex justify-content-center">
+      <div
+        style={{ wordBreak: 'break-word' }}
+        onClick={() => setSelectedLesson(i)}
+      >
+        <h4 style={{ margin: 'auto', wordBreak: 'break-word' }}>{obj.title}</h4>
+      </div>
+    </div>
+  </div>
+)
+
+export const AdminLessonsSideBar: React.FC<SideBarLessonProps> = ({
+  lessons,
+  setSelectedLesson
+}) => {
+  let lessonListData = lessons || []
```

Comment:
```
const lessonListData = lessons || []
lessonListData.push({title: '....'})
const lessonList = .....
```

Better to use `const`, good stuff for refactoring the original comment.

Reply:
since this is not really reusable and not too long, does it make sense to put this inside `pages/admin/lessons.tsx`?